### PR TITLE
storage(android): update getURL options

### PR DESCRIPTION
**Pull request by phantumcode**
#### Description of changes:
Update Android Storage getURL with options to validate object existence with changes from the following amplify-android [PR 2848](https://github.com/aws-amplify/amplify-android/pull/2848)

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [x] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] Swift
- [x] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [x] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [x] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [x] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [x] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [x] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/pages/[platform]/build-a-backend/storage/download-files/index.mdx [hunk 1]
```diff
@@ -151,6 +151,36 @@ RxAmplify.Storage.getUrl(StoragePath.fromString("public/example")).subscribe(
 
 </Block>
 </BlockSwitcher>
+
+### Check the existence of a file
+
+When creating a downloadable URL, you can choose to check if the file exists by setting `validateObjectExistence` to
+`true` in `AWSS3StorageGetPresignedUrlOptions`. If the file is inaccessible or does not exist, a `StorageException` is thrown.
+This allows you to check if an object exists during generating the presigned URL, which you can then use to download
```
Comment (vincetran) [inline src/pages/[platform]/build-a-backend/storage/download-files/index.mdx:159]: > This allows you to check if an object exists **during** generating the presigned URL

Would reword to "This allows you to check if an object exists **when** generating the presigned URL"

File: src/pages/[platform]/build-a-backend/storage/download-files/index.mdx [hunk 2]
```diff
@@ -151,6 +151,36 @@ RxAmplify.Storage.getUrl(StoragePath.fromString("public/example")).subscribe(
...
+that object.
+
+
+```kotlin
```
Comment (vincetran) [inline src/pages/[platform]/build-a-backend/storage/download-files/index.mdx:163]: Should we include a block switcher to show a code example for java, kotlin coroutine, and rxjava?

Comment (phantumcode) [reply src/pages/[platform]/build-a-backend/storage/download-files/index.mdx:181]: good suggestion, block switcher added.

File: src/pages/[platform]/build-a-backend/storage/download-files/index.mdx (hunk 1 above)
Comment (phantumcode) [reply src/pages/[platform]/build-a-backend/storage/download-files/index.mdx]: updated per suggestion.

**Review by hdworld11 — APPROVED** [review]

**Review by hbuchel — APPROVED** [review]
